configs/configload/copy_dir.go

```diff
@@ -59,6 +59,20 @@ func copyDir(dst, src string) error {
 			return nil
 		}
 
+		// If the current path is a symlink, recreate the symlink relative to
+		// the dst directory
+		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+			// the original symlink target
+			target, err := filepath.EvalSymlinks(path)
+			if err != nil {
+				return err
+			}
+
+			newTarget, err := filepath.Rel(src, target)
```

Comment:
It occurred to me reading this that the symlink target path in the _source_ directory here is presumably already pointing at a reasonable location relative to the source directory, and we're creating an exact copy of that directory so the exact same target should in principle work in the target too, without the need to use `EvalSymlinks` or `Rel` here to rewrite it.

Could we do the following simpler sequence here instead?

```go
target, err := os.Readlink(path)
if err != nil {
    return err
}

return os.Symlink(target, dstPath)
```

:thinking: Possibly I'm missing a situation where that doesn't work out...

Reply:
I think most tools opt to copy the symlink exactly, and leave it broken if it was pointing to something outside the tree being copied. 

I agree that just finding the target and making a new symlink is sufficient here.

Reply:
Yes, absolutely! That's what I did originally, but I was concerned that it would be insufficient - I might have been inventing potential issues. Thanks!